pkg/build/api/v1/types.go

```diff
@@ -52,8 +52,73 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64 `json:"completionDeadlineSeconds,omitempty"`
+
+	// Info is a placeholder that can store multiple reasons why a build was
+	// triggered. It should cover for a future implementation of level triggered
+	// builds where you can multiple reasons why a build was triggered.
+	Info []*BuildTriggerInfo `json:"info,omitempty"`
```

Comment:
not exactly, like i said above i will push the updates a bit later, but i wouldn't mind the suggestions so that i don't have to refactor. the Info is now renamed to Causes.
please also see this comment from above https://github.com/openshift/origin/pull/7518#discussion_r56700320